### PR TITLE
test(desktop): add coverage for Agent Studio orchestration hooks

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
- add direct tests for Agent Studio orchestration hooks that are role/session/permission sensitive
- cover document update orchestration, task hydration transitions, and permission reply state/error flows
- protect deep-link and session-context behavior with targeted edge-case assertions and regression paths

## Testing
- bun test "src/pages/use-agent-studio-documents.test.tsx" "src/pages/use-agent-studio-task-hydration.test.tsx" "src/pages/use-agent-session-permission-actions.test.tsx"
- bun run --filter @openducktor/desktop typecheck
- bun run --filter @openducktor/desktop lint
- bun run --filter @openducktor/desktop test

### CODE REVIEW

File: apps/desktop/src/pages/use-agent-studio-task-hydration.test.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    }
+  });
+
+  test("marks a task hydrated even when loading sessions rejects", async () => {
+    const deferred = createDeferred<void>();
+    const loadAgentSessions = mock(async () => {
+      try {
+        await deferred.promise;
+      } catch {}
+    });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/use-agent-studio-task-hydration.test.tsx]: Fixed in 20b50bb. The rejection-path test now returns a genuinely rejecting load promise and explicitly validates that rejection before asserting hydration state. It also safely consumes the `.finally` chain to avoid unhandled rejection noise in test runtime, so the rejection branch is now both deterministic and meaningful.

**Comment by Maxsky5** [comment]
@gemini-code-assist[bot] Thanks for the summary. I reviewed all feedback and the only actionable item was the inline rejection-path concern on `use-agent-studio-task-hydration.test.tsx`, which is now fixed in 20b50bb with full validation (build, full tests, lint). No additional code changes were required from this informational summary comment.
